chore: deploy test

cmwylie19 opened the pull request:
## Description

unit test for deploy

## Related Issue

Fixes #2396 
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

cmwylie19 committed:
chore: deploy test

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
chore: deploy test

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2396

samayer12 reviewed (changes requested):
Our other CLI unit tests execute with `program.parseAsync()` and trigger code that way. The current set of tests export previously unexported functions that run during `.action()` and ties us to implementation details.

We should generate coverage for our branching logic within 130-160 by following the `parseAsync()` test pattern we use in `kfc.test.ts`, `monitor.test.ts`, `uuid.test.ts`, or `update/index.test.ts`

cmwylie19 committed:
chore: create new test for program.parseAsync due to mocks

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
Merge branch 'main' into 2396

cmwylie19 commented:
> in `kfc.test.ts`, `monitor.test.ts`, `uuid.test.ts`, or `update/index.test.ts`

Due to needing to test the functions in deploy and also needing to mock them to test `command.parseAsync` i needed to separate this test into another file

cmwylie19 committed:
chore: parseAsync from user

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

samayer12 committed:
2396 review (#2518)

## Description

This PR breaks the deploy details into smaller, more testable chunks. It
also updates mocking using during tests.

## Related Issue

Relates to #2396

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

samayer12 committed:
Merge branch 'main' into 2396

samayer12 reviewed (approved):
Discussed changes via slack
